www/js/intro.js

```diff
@@ -23,6 +23,20 @@ angular.module('emission.intro', ['emission.splash.startprefs',
 .controller('IntroCtrl', function($scope, $state, $window, $ionicSlideBoxDelegate,
     $ionicPopup, $ionicHistory, ionicToast, $timeout, CommHelper, StartPrefs, UpdateCheck, $translate, i18nUtils) {
 
+  $scope.surveyState = { 
+    isValid: true, 
+    schema: {'fname':'Jenna', 'lname':'Ruzekowicz'},
+    result: {'fname':'', 'lname':''} 
+  };
+  $scope.changeresult = function(field, value) {
+    if (field == 'fname') {
+      $scope.surveyState.result.fname = value
+    }
+    if (field == 'lname') {
+      $scope.surveyState.result.lname = value
+    }
```

Comment:
Angular doesn't have immutable objects, so would this work? Seems a bit cleaner.
It's OK if it doesn't work, though, because we can just set up the final result to be something like 
`$scope.surveyState.result.theResult` or something.


```suggestion
  $scope.changeresult = function(value) {
      $scope.surveyState.result = value
```

Reply:
Yes! I have been thinking of ways to simplify this and will give some attempts today. Then I will move to http and writing!!